wpa_supplicant/wpa_cli.c

```diff
@@ -307,14 +307,20 @@ static int wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd)
 	}
 
     repl_buf = wpa_supplicant_ctrl_iface_process (wpa_s, (char *) cmd, &len);
-    
-	if (len > 0 && os_strcmp(repl_buf, "OK") != 0)
+
+	if (len == 2 && os_strcmp(repl_buf, "OK") != 0)
 	   printf ("Reply: %s\n", repl_buf);
-	
-	if (len > 0 && repl_buf)
-	    os_free(repl_buf);
+	else if (len == 5 && os_strcmp(repl_buf, "FAIL") == 0) {
```

Comment:
Should len be 4 here?

Reply:
the string is `FAIL\n` so it should be 5, we are comparing only first 4, I can convert this to 5 and add \n.

Reply:
Ok..I guess then the "OK" case above also needs to have a similar change?